src/librustc/infer/opaque_types/mod.rs

```diff
@@ -1155,6 +1157,15 @@ impl<'a, 'tcx> Instantiator<'a, 'tcx> {
         );
         debug!("instantiate_opaque_types: ty_var={:?}", ty_var);
 
+        for predicate in &bounds.predicates {
```

Comment:
We could make a method returning `-> bool` here and use `if stuff.obligation_references_error() { return ty_var; }` here.

Reply:
I didn't want to hide the `O(n)` behavior of the added code so that I could maybe nerd-snipe someone to point me a place where we could do the same with no added time ^_^

Reply:
Can't you just leave a doc comment about the `O(n)` complexity? This is in diagnostics code anyways so what does `O(n)` matter?

Reply:
Because it it needs to go over the predicates vec twice instead of once for _every_ case. This is to add the obligations, which may or may not be met.